File: server/api/manual_job.py
"""Manual job API endpoints."""
from server.core.auth import require_auth

from fastapi import APIRouter, Depends, HTTPException, Query

from server.models.manual_job import (
    ManualJob,
    ManualJobCreate,
    ManualJobDeleteRequest,
    ManualJobListResponse,
    ManualJobStatus,
)
from server.services.manual_job_service import ManualJobService

router = APIRouter(prefix="/api/manual-jobs", tags=["manual-jobs"], dependencies=[Depends(require_auth)])


def get_service() -> ManualJobService:
    """Dependency injection for ManualJobService."""
    return ManualJobService()


@router.post("", response_model=ManualJob)
async def create_job(
    job: ManualJobCreate,
    service: ManualJobService = Depends(get_service),
) -> ManualJob:
    """Create a new manual job."""
    return await service.create_job(job)


@router.get("", response_model=ManualJobListResponse)
async def list_jobs(
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    search: str | None = Query(None),
    status: ManualJobStatus | None = Query(None),
    service: ManualJobService = Depends(get_service),
) -> ManualJobListResponse:
    """List manual jobs with pagination."""
    offset = (page - 1) * page_size
    jobs, total = await service.list_jobs(
        limit=page_size,
        offset=offset,
        search=search,
        status=status,
    )
    return ManualJobListResponse(jobs=jobs, total=total)


@router.get("/{job_id}", response_model=ManualJob)
async def get_job(
    job_id: int,
    service: ManualJobService = Depends(get_service),
) -> ManualJob:
    """Get a manual job by ID."""
    job = await service.get_job(job_id)
    if job is None:
        raise HTTPException(status_code=404, detail="手动任务不存在")
    return job


@router.delete("")
async def delete_jobs(
    request: ManualJobDeleteRequest,
    service: ManualJobService = Depends(get_service),
) -> dict:
    """Delete manual jobs by IDs."""
    deleted = await service.delete_jobs(request.ids)
    return {"deleted": deleted}

File: web/src/api/manual-job.ts
import api from './index'
import type {
  ManualJob,
  ManualJobCancelResult,
  ManualJobCreate,
  ManualJobListResponse,
  ManualJobStatus,
} from './types'

/**
 * 手动任务 API
 */
export const manualJobApi = {
  /**
   * 创建手动任务
   */
  async create(data: ManualJobCreate): Promise<ManualJob> {
    const response = await api.post<ManualJob>('/manual-jobs', data)
    return response.data
  },

  /**
   * 获取任务列表
   */
  async list(params: {
    page?: number
    page_size?: number
    search?: string
    status?: ManualJobStatus | null
  } = {}): Promise<ManualJobListResponse> {
    const response = await api.get<ManualJobListResponse>('/manual-jobs', { params })
    return response.data
  },

  /**
   * 获取单个任务
   */
  async get(id: number): Promise<ManualJob> {
    const response = await api.get<ManualJob>(`/manual-jobs/${id}`)
    return response.data
  },

  /**
   * 批量删除任务
   */
  async delete(ids: number[]): Promise<{ deleted: number }> {
    const response = await api.delete<{ deleted: number }>('/manual-jobs', { data: { ids } })
    return response.data
  },

  /** 取消扫描，并等待其未完成的刮削子任务安全收尾。 */
  async cancel(id: number): Promise<ManualJobCancelResult> {
    const response = await api.post<ManualJobCancelResult>(
      `/manual-jobs/${id}/cancel`,
      undefined,
      { timeout: 0 },
    )
    return response.data
  },
}
